packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -215,9 +215,16 @@ export function getBeaconStateApi({
             }
             balances.push({index: id, balance: state.balances.get(id)});
           } else {
-            const index = headState.epochCtx.pubkey2index.get(fromHex(id));
-            if (index != null && index <= state.validators.length) {
-              balances.push({index, balance: state.balances.get(index)});
+            if (id.startsWith("0x")) {
```

Comment:
we might be able to reuse `getStateValidatorIndex` here https://github.com/ChainSafe/lodestar/blob/f3e21c76794869cadedccce10ca02401e150351a/packages/beacon-node/src/api/impl/beacon/state/utils.ts#L150